WeatherApp/WeatherApp/ViewController/ViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ViewController.swift
+//  WeatherApp
+//
+//  Created by Thomas Feng on 9/24/19.
+//  Copyright © 2019 Thomas Feng. All rights reserved.
+//
+
+import UIKit
+import Log
+
+class ViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadWeatherData()
+    }
+    
+    @IBOutlet var collectionView: UICollectionView!
+    
+    let reuseIdentifier = "forecastCell"
+    var forecasts = [Forecast]()
+    let log = Logger()
+    
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return forecasts.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+            let forecast = forecasts[indexPath.item]
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! WeatherCell
+        cell.tempLabel.text = String(forecast.temperatureHigh)
+        cell.imageView.image = forecast.image
+        cell.date.text = forecast.date
+            return cell
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let screenSize = UIScreen.main.bounds
+        let screenWidth = screenSize.width
+        let cellDimension = screenWidth / 2.1
+        return CGSize(width: cellDimension, height: cellDimension)
+    }
+    
+    func loadWeatherData() {
+        WeatherStore.getForecast { (result) in
+            switch result {
+            case let .success(data):
+//                self.log.trace(data.daily)
+                self.forecasts = data.daily.data
+                print(self.forecasts)
+            case let .failure(error):
+                self.log.trace(error)
+            }
+            DispatchQueue.main.async {
+                self.collectionView.reloadData()
+            }
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        self.performSegue(withIdentifier: "showDetail", sender: self)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        let forecastDetailViewController = segue.destination as! ForecastDetailViewController
+        let indexPath = collectionView.indexPathsForSelectedItems?.first
+        forecastDetailViewController.forecast = forecasts[indexPath!.item]
+    }
```

Comment:
Option 1:
I suggested a coalescing operator above, but in this case, when loading a new VC, you probably want the whole function to exit instead and present a fatalError(). 

My reasoning: In general, if your use gets to this point and we are preparing for segue, but this specific case presents itself where the actual datasource is either `nil` or somehow not addressable, it is going to be a mistake on the developer's part, not the user, meaning: if the UI is available for the user to interact with on a `UICollection View`, but it is not actually backed up by any datasource (which is what this block looks for) then there is an implementation mistake and you should use `fatalError()` so that you don't keep moving forward without addressing this on the development side.
```
guard let indexPath = collectionView.indexPathsForSelectedItems?.first else {
    print("some error message")
    return
}
```

Option 2: Refactoring the whole flow
If we were to totally re-think the premise of this block, I would probably eliminate the need to call the index of `collectionView` at all by:
- removing the `prepareForSegue` function
- programmatically instantiating the `forecastDetailViewController` on line 63
- setting the variable on the new VC instance directly
- pushing the `forecastDetailViewController` directly on to the current VC using `present()`, or by using the `NavigationController` (whichever you're working with in that context)